Demo/Demo/SwiftUIComponents/CardPaymentViews/CardOrderActionButton.swift

```diff
@@ -0,0 +1,49 @@
+import SwiftUI
+
+struct CardOrderActionButton: View {
+
+    let intent: String
+    let orderID: String
+    let selectedMerchantIntegration: MerchantIntegration
+
+    @ObservedObject var cardPaymentViewModel: CardPaymentViewModel
+
+    var body: some View {
+        ZStack {
+            Button("\(intent)") {
+                if intent == "CAPTURE" {
+                    Task {
+                        do {
+                            try await cardPaymentViewModel.captureOrder(
+                                orderID: orderID,
+                                selectedMerchantIntegration: selectedMerchantIntegration
+                            )
+                            print("Order Captured. ID: \(cardPaymentViewModel.state.capturedOrder?.id ?? "")")
+                        } catch {
+                            print("Error capturing order: \(error.localizedDescription)")
+                        }
+                    }
+                } else {
+                    Task {
+                        do {
+                            try await cardPaymentViewModel.authorizeOrder(
+                                orderID: orderID,
+                                selectedMerchantIntegration: selectedMerchantIntegration
+                            )
+                            print("Order Authorized. ID: \(cardPaymentViewModel.state.authorizedOrder?.id ?? "")")
+                        } catch {
+                            print("Error authorizing order: \(error.localizedDescription)")
+                        }
+                    }
+                }
+            }
+            .buttonStyle(RoundedBlueButtonStyle())
+            .padding()
+            if case .loading = cardPaymentViewModel.state.authorizedOrderResponse {
+                CircularProgressView()
+            } else if case .loading = cardPaymentViewModel.state.capturedOrderResponse {
```

Comment:
I think we can update this to the same if logic in your other PR

Reply:
I remembered that I did this because these states have associated values so I can't use equality, that's why I opted for pattern matching.

Reply:
I'm not sure I follow - on the other branch I tested in locally and it appeared to work as expected. Are you seeing different behavior on an error?

Reply:
fixed here: 562035932a6772edc0522c28cb3a162b29a08277